### PR TITLE
Rename EmitInfantryOnSell and remove default actor type.

### LINKED ISSUE #15077: Remove default ActorTypes: from EmitInfantryOnSell:
I was told that such kind of defaults have been purged 2 years ago so in this case having `ActorTypes: e1` as a default of `EmitInfantryOnSell:` is inconsistent and should therefore be removed.

### PR DESCRIPTION
**Pull request by ghost**
To fix #15077 and probably close #15161. I tested adding "ActorTpes" for the case that only the default is used (our mods do not use the default anywhere) and it seemed to work.

### CODE REVIEW

**Comment by reaperrr** [comment]
@lawando I guess all the oramaps were showing up as having been "updated" by the update rule, but for oramap files that's actually a false alarm caused by the tool touching the file, even when no changes are applied.
Any maps saved as .oramap never contain custom rules (in our official mods, at least), so it's safe to discard the "updated" oramaps from this PR.

**Comment by lawando** [comment]
Updated.

File: OpenRA.Mods.Common/UpdateRules/Rules/RenameEmitInfantryOnSell.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+    public class RenameEmitInfantryOnSell : UpdateRule
+    {
+        public override string Name { get { return "EmitInfantryOnSell renamed to SpawnActorOnSell"; } }
+        public override string Description
+        {
+            get
+            {
+                return "The EmitInfantryOnSell trait was renamed to SpawnActorOnSell and the default \n" +
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/UpdateRules/Rules/RenameEmitInfantryOnSell.cs]: You don't need the line-ending spaces in the output.

**Review by GraionDilach — DISMISSED** [review]
:+1:

**Comment by GraionDilach** [comment]
Needs a rebase now.

**Comment by lawando** [comment]
Rebased.

**Comment by reaperrr** [comment]
`ActorTypes` still has `e1` as internal default. When you remove it, don't forget to remove the reference to it in the description as well.
I also think it then makes sense to add [FieldLoader.Require] to the property, to ensure that modders don't accidentally forget to define any actors.

**Comment by pchote** [comment]
@reaperrr I think the idea was to leave that part to #15161 so as to not close a contributors first PR

**Comment by lawando** [comment]
Updated, removed the default actor type.

**Comment by reaperrr** [comment]
Needs another rebase now, sorry.

**Comment by reaperrr** [comment]
One last thing I unfortunately only noticed now: unlike `SpawnActorOnDeath`, this trait can spawn multiple actors at once, so `SpawnActor**s**OnSell` would be more accurate.

Regarding the discussion on IRC about ActorTypes vs. Actors: We have some cases of both and not really decided on a convention yet, so maybe it's better to leave it as it is, and then later fix all property names across all affected traits in one fell swoop once we've decided on a convention.

**Comment by lawando** [comment]
Updated.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
I consider Graion's +1 still valid since the changes made since his review were minor, so LGTM.

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ed12e293ef1ac8ebe2dfb2102d4d6e6b8660836e...506b83167b8da17dc0f94eeba345cdb718cd1876)

**Pull request merged**